backend/kale/processors/nbprocessor.py

```diff
@@ -664,62 +682,156 @@ def dependencies_detection(self, imports_and_functions: str = ""):
                 marshal_candidates = astutils.get_marshal_candidates(
                     anc_source)
                 outs = ins_left.intersection(marshal_candidates)
-                # Remove the ins that have already been assigned to an ancestor
-                ins_left.difference_update(outs)
-                # Include free variables
-                to_remove = set()
+                print(f"Anc step {anc} marshal candidates inside dependency detect: {marshal_candidates}")
+                print(f"Anc step {anc} outs inside dependency detect: {outs}")
+                # # Remove the ins that have already been assigned to an ancestor
```

Comment:
@Amrit27k could you explain a bit the changes that you applies here? It seems like you introduced some new logic.

It seems like you wanted to keep the original code (what you left as commented code) but below modified it. I think that if you remove the commented code block, then the diffi should become easier to read and easier to compare with the previous version.

all-in-all, it would be great if you could describe how the new logic differs from the previous one